msteams-platform/toolkit/debug-teams-app-visual-studio.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Debug your Teams app for Visual Studio
+description: In this module, learn how to debug your Teams app locally in Visual Studio using Teams Toolkit
+ms.author: surbhigupta
+ms.localizationpriority: high
+ms.topic: overview
+ms.date: 03/21/2022
+---
+
+# Debug your Teams app locally using Visual Studio
+
+Teams Toolkit helps you to debug and preview your Teams app locally. Debug is the process of building, checking, detecting, and correcting issues or bugs to ensure the program runs successfully. Visual Studio allows you to debug tab, bot, message extension. Teams Toolkit supports the following debug features:
+
+* Prepare Teams app dependencies
+* Start debugging
+* Toggle breakpoints
+* Hot reload
+* Stop debugging
+
+During the debug process, Teams Toolkit automatically starts app services, launches debuggers, and side loads the Teams app. The Teams app is available for preview in Teams web client locally after debugging. You can also customize debug settings to use your bot endpoints, or environment variables to load your configured app.
+
+## Prerequisites
+
+* Install the latest version of Teams Toolkit for Visual Studio.
+* To run / debug a Bot or Message Extension app, you'll need to set up [Ngrok](https://ngrok.com/) first.
+Ngrok is used to forward external messages from Azure Bot Framework to your local machine.
+
+## Debug your app locally
+
+### Set up Ngrok (Only for Bot and Message Extension app)
+
+Use a Command Prompt to run this command:
+
+```
+ngrok http 5130.
+```
+
+### Set up your Teams Toolkit
+
+Perform the following steps to debug your app after you create a new app using the Teams Toolkit:
+
+1. In Visual Studio Solution Explorer, right click on your project file and select "Teams Toolkit"
```

Comment:
thought we should agree on the style. I noticed that we use double quote here indicating the button that users should click. In some other files, we use other styles like bold.

Reply:
As per MSTP:
Right-click is used with hyphen
Any UI element is highlighted in bold.
We will make the change.